### PR TITLE
Add Kotlin code for "headless-js-android" documentation

### PR DESCRIPTION
**Pull request by rokkoo**
3018

### CODE REVIEW

**Comment by rokkoo** [comment]
@cortinico @mdvacca

File: docs/headless-js-android.md [hunk 1]
```diff
@@ -26,12 +29,16 @@ module.exports = async (taskData) => {
 
 You can do anything in your task such as network requests, timers and so on, as long as it doesn't touch UI. Once your task completes (i.e. the promise is resolved), React Native will go into "paused" mode (unless there are other tasks running, or there is a foreground app).
 
-## The Java API
+## The Native API
```
Comment (cortinico) [inline docs/headless-js-android.md:32]:
```suggestion
## The Platform API
```

File: docs/headless-js-android.md [hunk 2]
```diff
@@ -57,6 +64,39 @@ public class MyTaskService extends HeadlessJsTaskService {
...
+        if (extras != null) {
+            return HeadlessJsTaskConfig(
+                "SomeTaskName",
+                Arguments.fromBundle(extras),
+                5000, // timeout for the task
+                false // optional: defines whether or not  the task is allowed in foreground.
+                // Default is false
+                )
+        }
+        return null
```
Comment (cortinico) [inline docs/headless-js-android.md:91]: Let's convert this to use `return intent.extras?.let { ... }`

File: docs/headless-js-android.md [hunk 3]
```diff
@@ -77,12 +120,32 @@ service.putExtras(bundle);
 getApplicationContext().startService(service);
 ```
 
+</TabItem>
+<TabItem value="kotlin">
+
+```kotlin
+val service = Intent(getApplicationContext(), MyTaskService::class.java)
```
Comment (cortinico) [inline docs/headless-js-android.md:127]:
```suggestion
val service = Intent(applicationContext, MyTaskService::class.java)
```

The IDE should suggest you to simplify here. Can you confirm?

File: docs/headless-js-android.md [hunk 4]
```diff
@@ -98,6 +161,22 @@ return new HeadlessJsTaskConfig(
...
+<TabItem value="kotlin">
+
+```kotlin
+val retryPolicy: HeadlessJsTaskRetryPolicy =
+    LinearCountingRetryPolicy(
+        3, // Max number of retry attempts
+        1000 // Delay between each retry attempt
+    )
+
+return HeadlessJsTaskConfig('SomeTaskName', Arguments.fromBundle(extras), 5000, false, retryPolicy)
```
Comment (cortinico) [inline docs/headless-js-android.md:174]: Single quotes are not allowed in Kotlin

File: docs/headless-js-android.md [hunk 5]
```diff
@@ -183,13 +277,124 @@ public class NetworkChangeReceiver extends BroadcastReceiver {
...
+            if(actNw == null) {
+                return false;
+            }
+
+            if(actNw.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) || actNw.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) || actNw.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+                return true;
+            }
+
+            return false;
+        }
```
Comment (cortinico) [inline docs/headless-js-android.md:300]: Why is this needed?

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 6]
```diff
@@ -26,12 +29,16 @@ module.exports = async (taskData) => {
 
 You can do anything in your task such as network requests, timers and so on, as long as it doesn't touch UI. Once your task completes (i.e. the promise is resolved), React Native will go into "paused" mode (unless there are other tasks running, or there is a foreground app).
 
-## The Java API
+## The Native API
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:32]:
```suggestion
## The Platform API
```

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 7]
```diff
@@ -57,6 +64,39 @@ public class MyTaskService extends HeadlessJsTaskService {
...
+
+class MyTaskService : HeadlessJsTaskService() {
+    override fun getTaskConfig(intent: Intent): HeadlessJsTaskConfig? {
+        val extras: Bundle? = intent.extras
+        if (extras != null) {
+            return HeadlessJsTaskConfig(
+                "SomeTaskName",
+                Arguments.fromBundle(extras),
+                5000, // timeout for the task
+                false // optional: defines whether or not  the task is allowed in foreground.
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:87]:
```suggestion
                false // optional: defines whether or not the task is allowed in foreground.
```

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 8]
```diff
@@ -57,6 +64,39 @@ public class MyTaskService extends HeadlessJsTaskService {
...
+                // Default is false
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:88]: Can we put this in the line above?

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 9]
```diff
@@ -57,6 +64,39 @@ public class MyTaskService extends HeadlessJsTaskService {
...
+import android.os.Bundle
+import com.facebook.react.HeadlessJsTaskService
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.jstasks.HeadlessJsTaskConfig
+
+class MyTaskService : HeadlessJsTaskService() {
+    override fun getTaskConfig(intent: Intent): HeadlessJsTaskConfig? {
+        val extras: Bundle? = intent.extras
+        if (extras != null) {
+            return HeadlessJsTaskConfig(
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:83]: Same comments as above (use `?.let{}`)

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 10]
```diff
@@ -98,6 +161,22 @@ return new HeadlessJsTaskConfig(
...
+<TabItem value="kotlin">
+
+```kotlin
+val retryPolicy: HeadlessJsTaskRetryPolicy =
+    LinearCountingRetryPolicy(
+        3, // Max number of retry attempts
+        1000 // Delay between each retry attempt
+    )
+
+return HeadlessJsTaskConfig('SomeTaskName', Arguments.fromBundle(extras), 5000, false, retryPolicy)
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:174]: Single quotes are not allowed in Kotlin (for Strings)

**Review by cortinico — COMMENTED** [review]
Thanks for sending this over 👍 Left several comments that needs to be addressed before we can merge this

File: docs/headless-js-android.md (hunk 3 above)
Comment (rokkoo) [reply docs/headless-js-android.md:127]: Yes! You're right, thanks!

File: docs/headless-js-android.md (hunk 4 above)
Comment (rokkoo) [reply docs/headless-js-android.md:174]: Wow.. what a mistake, sorry and thanks for the review!

File: docs/headless-js-android.md (hunk 5 above)
Comment (rokkoo) [reply docs/headless-js-android.md:300]: `getActiveNetworkInfo` is deprecated,  so we can use it in all APIs below 29.

https://developer.android.com/reference/android/net/NetworkInfo

One of the proposed on the docs is to use `getNetworkCapabilities`

https://developer.android.com/reference/android/net/ConnectivityManager#getNetworkCapabilities(android.net.Network)

File: website/versioned_docs/version-0.68/headless-js-android.md (hunk 8 above)
Comment (rokkoo) [reply website/versioned_docs/version-0.68/headless-js-android.md:88]: We can, but `ktfmt` move to the next line :(

**Comment by rokkoo** [comment]
> Let me know if you need further support @rokkoo

Thanks @cortinico for the review!

File: website/versioned_docs/version-0.68/headless-js-android.md (hunk 8 above)
Comment (cortinico) [reply website/versioned_docs/version-0.68/headless-js-android.md:86]: Let's stick with ktfmt's stlye then

File: docs/headless-js-android.md [hunk 11]
```diff
@@ -57,6 +64,36 @@ public class MyTaskService extends HeadlessJsTaskService {
...
+class MyTaskService : HeadlessJsTaskService() {
+    override fun getTaskConfig(intent: Intent): HeadlessJsTaskConfig? {
+        return intent.extras?.let {
+            HeadlessJsTaskConfig(
+                "SomeT  askName",
+                Arguments.fromBundle(it),
+                5000, // timeout for the task
+                false // optional: defines whether or not  the task is allowed in foreground.
+                // Default is false
+                )
```
Comment (cortinico) [inline docs/headless-js-android.md:87]:
```suggestion
            )
```

File: docs/headless-js-android.md [hunk 12]
```diff
@@ -77,12 +117,32 @@ service.putExtras(bundle);
...
+
+```kotlin
+val service = Intent(applicationContext, MyTaskService::class.java)
+val bundle = Bundle()
+
+bundle.putString("foo", "bar")
+
+service.putExtras(bundle)
+
+getApplicationContext().startService(service)
```
Comment (cortinico) [inline docs/headless-js-android.md:131]:
```suggestion
applicationContext.startService(service)
```

File: website/versioned_docs/version-0.68/headless-js-android.md [hunk 13]
```diff
@@ -77,12 +117,32 @@ service.putExtras(bundle);
...
+
+```kotlin
+val service = Intent(applicationContext, MyTaskService::class.java)
+val bundle = Bundle()
+
+bundle.putString("foo", "bar")
+
+service.putExtras(bundle)
+
+getApplicationContext().startService(service)
```
Comment (cortinico) [inline website/versioned_docs/version-0.68/headless-js-android.md:131]:
```suggestion
applicationContext.startService(service)
```

**Review by cortinico — COMMENTED** [review]
Did another pass @rokkoo 
Left a couple of comments but we should be able to merge after this 👍

**Comment by rokkoo** [comment]
Thanks @cortinico!

**Review by cortinico — APPROVED** [review]
